youtube_dlc/postprocessor/embedthumbnail.py

```diff
@@ -83,47 +93,82 @@ def is_webp(path):
             self.to_screen('Adding thumbnail to "%s"' % filename)
             self.run_ffmpeg_multiple_files([filename, thumbnail_filename], temp_filename, options)
 
-        elif info['ext'] == 'mkv':
+        elif info['ext'] in ['mkv', 'mka']:
+            # streams = self.get_metadata_object(filename)['streams']
             options = [
                 '-c', 'copy', '-map', '0', '-dn', '-attach', thumbnail_filename,
-                '-metadata:s:t', 'mimetype=image/jpeg', '-metadata:s:t', 'filename=cover.jpg']
-
-            self.to_screen('Adding thumbnail to "%s"' % filename)
-            self.run_ffmpeg_multiple_files([filename], temp_filename, options)
-
-        elif info['ext'] in ['m4a', 'mp4']:
-            if not check_executable('AtomicParsley', ['-v']):
-                raise EmbedThumbnailPPError('AtomicParsley was not found. Please install.')
-
-            cmd = [encodeFilename('AtomicParsley', True),
-                   encodeFilename(filename, True),
-                   encodeArgument('--artwork'),
-                   encodeFilename(thumbnail_filename, True),
-                   encodeArgument('-o'),
-                   encodeFilename(temp_filename, True)]
-            cmd += [encodeArgument(o) for o in self._configuration_args(exe='AtomicParsley')]
+                # '-metadata:s:%s' % (len(streams)) - Why
```

Comment:
If the .info.json attachment from the add-metadata step is included at this point, then ffmpeg will freak out if you specify the subsequent `-metadata:s:t` args, because as it matches both attachments you would be marking a .json file as a thumbnail

I have no idea if it's possible to do something like `-metadata:s:t:1`, but I found explicitly using the stream number easier

Reply:
then, if the user runs the command again, the file gets attached multiple times. I have pushed a patch to fix it. Have a look

Reply:
Looks good to me